internal/n/crit.go

```diff
@@ -16,29 +16,29 @@ type EventCriteria func(s.Event) bool
 // arguments will accept all given events.
 func EAnd(crits ...EventCriteria) EventCriteria {
 	return func(ev s.Event) bool {
-		result := true
 		for _, crit := range crits {
-			result = result && crit(ev)
-			if !result {
-				return result
+			if !crit(ev) {
+				return false
 			}
 		}
-		return result
+		return true
 	}
 }
 
 // EOr joins a slice of EventCriteria with an "or" statement. A call without
-// arguments wil reject all given events.
+// arguments wil accept all given events.
 func EOr(crits ...EventCriteria) EventCriteria {
 	return func(ev s.Event) bool {
-		result := false
+		if len(crits) == 0 {
+			return true
```

Comment:
Do you really want this? An empty list later or'ed with another list will result in a true result even if all the explicit results are false. I think you had it right before.

Reply:
I think the issue is more that `EAnd` and `EOr` previously behaved differently inputs. I think they should behave the same and I would opt for whichever default value results in less events being dropped to be more friendly to mistakes.

Reply:
Ideally, we could have a type that guarantees a non-empty list, but we are not going to get that here. 

@jpb can you advocate for your comment here? I'm ambivalent the window is already broken when go is involved :man_shrugging:

Reply:
You know, the comment indicates the behavior. I find the previous behavior more consistent with my intuition of and/or, but having 3 truth values does not help, and we are stuck with the language. Leave it as is, if you like.